docarray/index/backends/milvus.py

```diff
@@ -0,0 +1,422 @@
+import re
+import uuid
+from collections import defaultdict
+from dataclasses import dataclass, field
+from typing import (
+    TYPE_CHECKING,
+    Any,
+    Dict,
+    Generator,
+    Generic,
+    List,
+    Optional,
+    Sequence,
+    Type,
+    TypeVar,
+    Union,
+    cast,
+)
+
+from docarray import BaseDoc, DocList
+from docarray.index.abstract import BaseDocIndex
+from docarray.index.backends.helper import _execute_find_and_filter_query
+from docarray.typing import AnyTensor
+from docarray.typing.id import ID
+from docarray.typing.tensor.abstract_tensor import AbstractTensor
+from docarray.utils._internal._typing import safe_issubclass
+from docarray.utils._internal.misc import (
+    import_library,
+    is_tf_available,
+    is_torch_available,
+)
+from docarray.utils.find import _FindResult, _FindResultBatched
+
+torch_available, tf_available = is_torch_available(), is_tf_available()
+
+if torch_available:
```

Comment:
why do we need to import these?

Reply:
That's because when we perform a vector search, it requires the input in Python list format. If the user's input tensor is in torch/tensorflow format, we need to perform a conversion.